PowerValidatedSolutions.psm1

```diff
@@ -41100,22 +41100,69 @@ Function Get-vRAIntegrationDetail {
 
     Try {
         $vraapiVersion = "apiVersion=" + (Get-vRAAPIVersion)
+        if ( $vraapiVersion ) {
         $uri = "https://$vraAppliance/iaas/api/integrations?$vraapiVersion"
-        $response = Invoke-RestMethod -Method Get -Uri $uri -Headers $vraHeaders
-        if ($getVCID) {
+        $response = Invoke-RestMethod -Method Get -Uri $uri -Headers $vraHeaders -SkipCertificateCheck
+        if ( $getVCID ) {
             ($response.content | Where-Object integrationType -Match $integrationType).customProperties.vcIds
         }
-        if ($getIntegrationID) {
+        if ( $getIntegrationID ) {
             if (($PsBoundParameters.ContainsKey("integrationType")) -and ($PsBoundParameters.ContainsKey("integrationName"))) {
                 (($response.content | Where-Object integrationType -Match $integrationType) | Where-Object name -Match "\b$integrationName\b").id
             }
         }
+        $vropsTrue = $response.content | Where-Object { $_.integrationType -eq "vrops" }
+            if ( $vropsTrue ) {
+                $vropsTrue
+            } else {
+                $vropsTrue
+                Write-Error "Unable to get the VMware Aria Automation API version, please execute Request-vRAToken to get the token and then run Get-vRAIntegrationDetail."
+            }
+        }
     } Catch {
         Write-Error $_.Exception.Message
     }
 }
 Export-ModuleMember -Function Get-vRAIntegrationDetail
 
+Function Get-vRAIntegrationDetailvROPSStatus {
+
+        .SYNOPSIS
+         Get an VMware Aria Operations integration status from VMware Aria Automation.
+
+        .DESCRIPTION
+        The Get-vRAIntegrationDetailvROPSStatus cmdlet returns the integration details between VMware Aria Operations and VMware Aria Automation.
+
+       .EXAMPLE
+        Get-vRAIntegrationDetailvROPSStatus -integrationType "vrops"
```

Comment:
Could this be simplified to `Get-vRAIntegrationDetailStatus`? Since you have the Param and ValidateSet it could be used to check the status of other integrations (eg orchestrator) too instead of a singular.

Would another option (maybe) be that you could extend `Get-vRAIntegrationDetailStatus` with a `-status` switch?